Jetpack SSO: Update footer to add shared details modal dialog

ebinnion opened the pull request:
In a design review, it was mentioned that we should display the shared details (information that is shared from WP.com to the remote Jetpack site when a user logs in via SSO) in a modal.

To do this, we needed to add the shared details to the JetpackSSO redux store and then we needed to modify the SSO UI to accommodate. This PR does both.

cc @rickybanister for design review and @lezama for code review.

For design review, let's keep in mind that we have addressed several issues in #5762. So, for this PR, we should focus on the look of the footer (with the log in button) and the modal.

cc @kevko for a review of the wording that we're using in the footer. Specifically, `By logging in you agree to share {{detailsLink}}details{{/detailsLink}} between WordPress.com and %(siteName)s.`

To test:

- Checkout `update/sso-add-shared-details` branch
- Make sure that you're on the latest Jetpack master
- Go to `$site/wp-admin`
- Click "Log in with WordPress.com" button (make sure you test with a user that is not connected to WP.com and with SSO module enabled)
- You should land on `wpcalypso.wordpress.com/jetpack/sso...`.
- Replace `wpcalypso.wordpress.com` with `calypso.localhost:3000`
- You should see something that looks similar to this:
![screen shot 2016-06-02 at 10 59 15 pm](https://cloud.githubusercontent.com/assets/1126811/15768365/bc07fd72-2915-11e6-98ac-c2f4fc304caf.png)

Here's what the modal looks like:

![screen shot 2016-06-02 at 10 50 08 pm](https://cloud.githubusercontent.com/assets/1126811/15768225/ae3674f4-2914-11e6-88ed-4aa235aca7ea.png)

![screen shot 2016-06-02 at 10 50 15 pm](https://cloud.githubusercontent.com/assets/1126811/15768232/b5fcffb4-2914-11e6-92ae-5129b3ca32ed.png)

![screen shot 2016-06-02 at 10 50 19 pm](https://cloud.githubusercontent.com/assets/1126811/15768235/b9305f78-2914-11e6-9f38-b55873dbb6e5.png)

rickybanister commented:
Let's make the 'close' button say 

`Got it`

Can we link 'share details' instead of just 'details'.

Also would love a review by @kevko and maybe @richardmuscat for wording/legality/etc.

ebinnion commented:
@rickybanister – I have updated the PR. Here are some new screenshots:

![screen shot 2016-06-03 at 4 48 55 pm](https://cloud.githubusercontent.com/assets/1126811/15793718/76b51750-29ab-11e6-9816-adffe297577b.png)
![screen shot 2016-06-03 at 4 48 52 pm](https://cloud.githubusercontent.com/assets/1126811/15793717/76b2f998-29ab-11e6-8824-5279e0b7fd01.png)

Also, I went ahead and rebased against master. So, at this point we should see all of the design feedback from our walkthrough coming together.

lezama commented on client/state/jetpack-connect/test/actions.js:
```diff
@@ -48,6 +48,19 @@ describe( 'actions', () => {
+			first_name: 'Lou',
+			last_name: 'Bucket',
+			display_name: 'bestbbqtester',
+			description: 'I like BBQ, a lot.',
```
😛

lezama commented:
Code looks 👍 to me

rickybanister commented:
🚢

ebinnion commented:
I rebased against master and remove a commented line of CSS. Going to ship.
